apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -25,18 +25,59 @@ import {
 } from '@novu/application-generic';
 import {
   CreateWorkflowDto,
+  DEFAULT_WORKFLOW_PREFERENCES,
   StepCreateDto,
   StepDto,
+  StepTypeEnum,
   StepUpdateDto,
   WorkflowCreationSourceEnum,
   WorkflowOriginEnum,
   WorkflowResponseDto,
+  WorkflowStatusEnum,
   WorkflowTypeEnum,
 } from '@novu/shared';
 import { UpsertWorkflowCommand } from './upsert-workflow.command';
 import { StepUpsertMechanismFailedMissingIdException } from '../../exceptions/step-upsert-mechanism-failed-missing-id.exception';
 import { toResponseWorkflowDto } from '../../mappers/notification-template-mapper';
 
+const workflowResponseDto: WorkflowResponseDto = {
+  id: 'my-workflow-FtfjFHJ9j11gGfds', // `slug` + `-{base62(ObjectId)}`
+  slug: 'my-workflow', // previously, this was `workflow.triggers[0].identifier`
+  name: 'My Workflow',
+  description: 'My Workflow Description',
+  active: true,
+  type: WorkflowTypeEnum.BRIDGE,
+  origin: WorkflowOriginEnum.NOVU_CLOUD,
+  steps: [
+    {
```

Comment:
I think ID should be always database id, always, no exceptions, slug should be the mix of the name and the abbriviation of the database id